src/trace_processor/importers/proto/winscope/protolog_message_decoder.h

```diff
@@ -62,13 +63,15 @@ class ProtoLogMessageDecoder : public Destructible {
   explicit ProtoLogMessageDecoder(TraceProcessorContext* context);
   virtual ~ProtoLogMessageDecoder() override;
 
-  static ProtoLogMessageDecoder* GetOrCreate(TraceProcessorContext* context) {
-    if (!context->protolog_message_decoder) {
-      context->protolog_message_decoder.reset(
-          new ProtoLogMessageDecoder(context));
+  static ProtoLogMessageDecoder* GetOrCreate(
```

Comment:
All of these GetOrCreate functions can now be removed now you've decoupled yourself from the context.